flink-table/flink-sql-client/src/main/java/org/apache/flink/table/client/gateway/local/LocalExecutor.java

```diff
@@ -544,6 +544,11 @@ public ProgramTargetDescriptor executeUpdate(String sessionId, String statement)
 			String statement) {
 		applyUpdate(context, context.getTableEnvironment(), context.getQueryConfig(), statement);
 
+		//Todo: we should refactor following condition after TableEnvironment has support submit job directly.
+		if (!statement.trim().matches("(INSERT\\s+INTO.*)")) {
```

Comment:
why does this PR have things to do with "INSERT INTO"?

Reply:
`applyUpdate` method indeed call the TableEnvironment.sqlUpdate method, an operation like ddl get executed in the current design logic while `insert into ` will be buffered until `execute` method gets called.
So now it's a little hack way to deal with "INSERT INTO".

Reply:
Looks quite hack to me, @danny0405 do you have any idea how to deal with this?